src/master-playlist-controller.js

```diff
@@ -542,7 +598,21 @@ export class MasterPlaylistController extends videojs.EventTarget {
       this.onEndOfStream();
     });
 
-    this.mainSegmentLoader_.on('earlyabort', () => {
+    this.mainSegmentLoader_.on('earlyabort', (event) => {
+      if (this.bufferWaterLineSelector) {
```

Comment:
In `segmentLoader` we used to abort the download during an `earlyabort`, but that was problematic because with the new switching algorithm at certain buffer values we do not want to force a switch. Ultimately I think that we should move most of the logic for what triggers `earlyabort` in here, but doing that in a backwards compatible way wasn't going to be possible without drastic effort.

Reply:
So, now we exclude the current playlist, but we may not switch away from it? Should we only exclude it in this case if we do end up switching away from it?

Reply:
We only temporarily exclude it below so that it isn't selected as the `nextPlaylist` in `selectPlaylist`. Then we remove the exclusion and make sure that we should switch to the next playlist before actually blacklisting it and switching. Somewhat confusing, I will add some comments here.

Reply:
Oh, I totally missed that. That makes sense. Comments would definitely be helpful.

Reply:
Also of note is that we always blacklist (which was the previous behavior) when not using the `bufferWaterLineSelector` option.

Reply:
Yup, that's excpted, thanks.